# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1530036745 14400
#      Tue Jun 26 14:12:25 2018 -0400
# Node ID b61f3500651fcbd7fecb0a3b3573453e11c3fd2a
# Parent  46836dd20f259ef674c4373036cddde55f97fc19
mkoctfile: fix check for using -o and -c with multiple source files (bug #54182)

* mkoctfile.in.cc (main): New variable, output_file_option.
Fix check for using -o and -c with multiple source files.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -573,16 +573,17 @@ main (int argc, char **argv)
   std::string objfiles, libfiles, octfile, outputfile;
   std::string incflags, defs, ldflags, pass_on_options;
   bool strip = false;
   bool no_oct_file_strip_on_this_platform = is_true ("%NO_OCT_FILE_STRIP%");
   bool compile_only = false;
   bool link_stand_alone = false;
   bool depend = false;
   bool printonly = false;
+  bool output_file_option = false;
 
   for (int i = 1; i < argc; i++)
     {
       std::string arg = argv[i];
 
       std::string file;
 
       if (ends_with (arg, ".c"))
@@ -666,16 +667,18 @@ main (int argc, char **argv)
         }
 #endif
       else if (arg == "-M" || arg == "-depend" || arg == "--depend")
         {
           depend = true;
         }
       else if (arg == "-o" || arg == "-output" || arg == "--output")
         {
+          output_file_option = true;
+
           if (i < argc-1)
             {
               arg = argv[++i];
               outputfile = arg;
             }
           else
             std::cerr << "mkoctfile: output filename missing" << std::endl;
         }
@@ -759,17 +762,17 @@ main (int argc, char **argv)
     }
 
   if (output_ext ==  ".mex"
       && vars["ALL_CFLAGS"].find ("-g") != std::string::npos)
     {
       defs += " -DMEX_DEBUG";
     }
 
-  if (compile_only
+  if (compile_only && output_file_option
       && (cfiles.size () + ccfiles.size () + f77files.size ()) > 1)
     {
       std::cerr << "mkoctfile: may not use -c and -o with multiple source files"
                 << std::endl;
       return 1;
     }
 
   std::string output_option;
